### PR TITLE
GFORMS-2152: Add a new property in destinations that calls an API sev…

### PR DESCRIPTION
**Pull request by lxol**
…eral times for a payload in an array

### CODE REVIEW

File: app/uk/gov/hmrc/gform/submission/handlebars/HandlebarsHttpApiSubmitter.scala [hunk 1]
```diff
@@ -52,30 +73,62 @@ class RealHandlebarsHttpApiSubmitter[F[_]](
...
-              )
-            }
-            httpClient.put(uri, body)
+            val bodies = destination.payload.fold(List(""))(body => parseAndProcessAsList(body))
+            callUntilError(
+              bodies.map((uri, _)),
+              httpClient.put,
+              modelTree.value.formId,
+              destination.id
+            )
```
Comment (josef-vlach) [inline app/uk/gov/hmrc/gform/submission/handlebars/HandlebarsHttpApiSubmitter.scala:94]: Can we keep handling of `destination.multiRequestPayload` handling login separated from current single payload implementation? This implementation is hard to follow.

Simple `if .. else` should do the trick
```
if (destination.multiRequestPayload) {
   // call parseAndProcessAsList with the assumption it is only for multiRequestPayload
   // Feel free to put code for multiRequestPayload into its own class etc.
} else {

  destination.method match {
    case HttpMethod.GET => httpClient.get(uri)
    case HttpMethod.POST => ... current code unchanged
}
```

**Review by josef-vlach — CHANGES_REQUESTED** [review]

File: app/uk/gov/hmrc/gform/submission/handlebars/HandlebarsHttpApiSubmitter.scala (hunk 1 above)
Comment (lxol) [reply app/uk/gov/hmrc/gform/submission/handlebars/HandlebarsHttpApiSubmitter.scala]: refactored to keep the current code unchanged when multiRequestPayload = false

**Review by josef-vlach — COMMENTED** [review]
I think there is quite a space for an improvement. My quick attempt is here : https://github.com/hmrc/gform/pull/940/commits/854abd58b1c78aa7096959d05fbefa79f7771789
If you are ok with the changes, I'll approve this.

**Comment by lxol** [comment]
> 

I simplified it even further. Now 3 arguments of `callUntilError` can be acquired from the scope

**Review by josef-vlach — APPROVED** [review]
